src/sys/unix/selector/io_source/edge_triggered.rs

```diff
@@ -0,0 +1,104 @@
+use crate::sys::Selector;
+use crate::{Interest, Registry, Token};
+use std::io;
+use std::os::unix::io::RawFd;
+
+struct InternalState {
+    selector: Selector,
+    token: Token,
+    interests: Interest,
+    fd: RawFd,
+    is_registered: bool,
+}
+
+impl Drop for InternalState {
+    fn drop(&mut self) {
+        if self.is_registered {
+            let _ = self.selector.deregister(self.fd);
+        }
+    }
+}
+
+pub(crate) struct IoSourceState {
+    inner: Option<Box<InternalState>>,
+}
+
+impl IoSourceState {
+    pub fn new() -> IoSourceState {
```

Comment:
Nit: same for below.

```suggestion
    pub(crate) fn new() -> IoSourceState {
```